libs/agno/agno/workflow/workflow.py

```diff
@@ -3636,6 +3637,68 @@ def to_dict(self) -> Dict[str, Any]:
             "session_id": self.session_id,
         }
 
+    def _calculate_session_metrics_from_workflow_metrics(self, workflow_metrics: WorkflowMetrics) -> Metrics:
+        """Calculate session metrics by aggregating all step metrics from workflow metrics"""
+        session_metrics = Metrics()
+        
+        # Aggregate metrics from all steps
+        for step_name, step_metrics in workflow_metrics.steps.items():
+            if step_metrics.metrics:
+                session_metrics += step_metrics.metrics
+        
+        session_metrics.time_to_first_token = None
+        
+        return session_metrics
+
+    def _get_session_metrics(self, session: WorkflowSession) -> Metrics:
+        """Get existing session metrics from the database"""
+        if session.session_data and "session_metrics" in session.session_data:
+            session_metrics_from_db = session.session_data.get("session_metrics")
+            if session_metrics_from_db is not None:
+                if isinstance(session_metrics_from_db, dict):
+                    return Metrics(**session_metrics_from_db)
+                elif isinstance(session_metrics_from_db, Metrics):
+                    return session_metrics_from_db
+        return Metrics()
+
+    def _update_session_metrics(self, session: WorkflowSession, workflow_run_response: WorkflowRunOutput):
+        """Calculate and update session metrics"""
+        # Get existing session metrics
+        session_metrics = self._get_session_metrics(session=session)
+        
+        # If workflow has metrics, convert and add them to session metrics
+        if workflow_run_response.workflow_metrics:
+            run_session_metrics = self._calculate_session_metrics_from_workflow_metrics(
```

Comment:
Can't you just add the workflow metrics itself? Isn't that already aggregated?

Reply:
no that's step wise.